### PR TITLE
Update installation steps for minimal deployment.

### PR DESCRIPTION
**Pull request by morvencao**

### CODE REVIEW

File: .spelling [hunk 1]
```diff
@@ -530,6 +530,7 @@ subresources
 re-patch
 re-applied
 CustomResourceDefinitions
+CutomResources
```
Comment (geeknoid) [inline .spelling]: Cutom -> Custom

File: content/docs/setup/kubernetes/minimal-install/index.md [hunk 2]
```diff
@@ -104,18 +103,20 @@ istio-pilot-58c65f74bc-2f5xn             1/1       Running   0          1m
 
 * For option 2, uninstall using Helm:
 
+> Uninstalling this chart does not delete Istio's registered CRDs. Istio by design expects
+> CRDs to leak into the Kubernetes environment. As CRDs contain all runtime configuration
+> data in CutomResources the Istio designers feel it is better to explicitly delete this
```
Comment (geeknoid) [inline content/docs/setup/kubernetes/minimal-install/index.md]: CutomResources -> custom resources

File: content/docs/setup/kubernetes/minimal-install/index.md [hunk 3]
```diff
@@ -104,18 +103,20 @@ istio-pilot-58c65f74bc-2f5xn             1/1       Running   0          1m
...
 
-    If your Helm version is less than 2.10.0, then you need to manually cleanup extra job resource before redeploy new version of Istio chart:
-
-    {{< text bash >}}
-    $ kubectl -n istio-system delete job --all
-    {{< /text >}}
+* If desired, run the following command to delete all CRDs:
 
-* If desired, delete the CRDs:
+> Warning: Deleting CRDs will delete any configuration that you have made to Istio.
```
Comment (geeknoid) [inline content/docs/setup/kubernetes/minimal-install/index.md]: configuration -> configuration changes

File: content/docs/setup/kubernetes/minimal-install/index.md [hunk 4]
```diff
@@ -69,7 +62,13 @@ to manage the lifecycle of Istio.
     $ helm init --service-account tiller
     {{< /text >}}
 
-1. Install Istio:
+1. Install the `istio-init` chart to bootstrap all the Istio's [CRDs]:
```
Comment (geeknoid) [inline content/docs/setup/kubernetes/minimal-install/index.md]: Why is CRDs in brackets?

File: content/docs/setup/kubernetes/minimal-install/index.md (hunk 3 above)
Comment (morvencao) [reply content/docs/setup/kubernetes/minimal-install/index.md]: Done.

File: content/docs/setup/kubernetes/minimal-install/index.md (hunk 4 above)
Comment (morvencao) [reply content/docs/setup/kubernetes/minimal-install/index.md]: Removed brackets.

File: content/docs/setup/kubernetes/minimal-install/index.md [hunk 5]
```diff
@@ -15,21 +15,15 @@ Refer to the [prerequisites](/docs/setup/kubernetes/quick-start/#prerequisites)
...
-1. Choose one of the following two
-**mutually exclusive** options described below.
+Choose one of the following two **mutually exclusive** options described below.
 
 ### Option 1: Install with Helm via `helm template`
 
 1. Render Istio's core components to a Kubernetes manifest called `istio-minimal.yaml`:
 
     {{< text bash >}}
+    $ cat install/kubernetes/namespace.yaml > $HOME/istio-minimal.yaml
```
Comment (geeknoid) [inline content/docs/setup/kubernetes/minimal-install/index.md]: Put @ on either end of the yaml file path, this will generate a link to the file on GitHub:

$ cat @install/kubernetes/namespace.yaml@ > $HOME/istio-minimal.yaml

Comment (morvencao) [reply content/docs/setup/kubernetes/minimal-install/index.md]: Great! done.

File: content/docs/setup/kubernetes/minimal-install/index.md [hunk 6]
```diff
@@ -15,21 +15,15 @@ Refer to the [prerequisites](/docs/setup/kubernetes/quick-start/#prerequisites)
...
-1. If using a Helm version prior to 2.10.0, install Istio's [Custom Resource Definitions](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/#customresourcedefinitions)
-via `kubectl apply`, and wait a few seconds for the CRDs to be committed in the kube-apiserver:
-
-    {{< text bash >}}
-    $ kubectl apply -f install/kubernetes/helm/istio/templates/crds.yaml
-    {{< /text >}}
-
-1. Choose one of the following two
-**mutually exclusive** options described below.
+Choose one of the following two **mutually exclusive** options described below.
```
Comment (rcaballeromx) [inline content/docs/setup/kubernetes/minimal-install/index.md]: How is the reader supposed to choose? On what should they base the decision?

**Comment by rcaballeromx** [comment]
I'm reviewing this in detail.
/hold

File: content/docs/setup/kubernetes/minimal-install/index.md [hunk 7]
```diff
@@ -69,7 +62,13 @@ to manage the lifecycle of Istio.
     $ helm init --service-account tiller
     {{< /text >}}
 
-1. Install Istio:
+1. Install the `istio-init` chart to bootstrap all the Istio's CRDs:
```
Comment (rcaballeromx) [inline content/docs/setup/kubernetes/minimal-install/index.md]: This is the new first mention of CRDs in the document, please expand the term and link to the definition as it was before:
```suggestion
1. Install the `istio-init` chart to bootstrap all the Istio's [Custom Resource Definitions or CRDs for short](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/#customresourcedefinitions):
```

File: content/docs/setup/kubernetes/minimal-install/index.md [hunk 8]
```diff
@@ -104,18 +103,20 @@ istio-pilot-58c65f74bc-2f5xn             1/1       Running   0          1m
 
 * For option 2, uninstall using Helm:
 
+> Uninstalling this chart does not delete Istio's registered CRDs. Istio by design expects
```
Comment (rcaballeromx) [inline content/docs/setup/kubernetes/minimal-install/index.md]:
```suggestion
> Uninstalling this chart does not delete Istio's registered CRDs. Istio, by design, expects
```

File: content/docs/setup/kubernetes/minimal-install/index.md [hunk 9]
```diff
@@ -104,18 +103,20 @@ istio-pilot-58c65f74bc-2f5xn             1/1       Running   0          1m
...
+> CRDs to leak into the Kubernetes environment. As CRDs contain all runtime configuration
+> data in custom resources the Istio designers feel it is better to explicitly delete this
```
Comment (rcaballeromx) [inline content/docs/setup/kubernetes/minimal-install/index.md]:
```suggestion
> data in custom resources, the Istio designers consider better to explicitly delete this
```

File: content/docs/setup/kubernetes/minimal-install/index.md [hunk 10]
```diff
@@ -104,18 +103,20 @@ istio-pilot-58c65f74bc-2f5xn             1/1       Running   0          1m
...
+> configuration rather then unexpectedly lose it.
```
Comment (rcaballeromx) [inline content/docs/setup/kubernetes/minimal-install/index.md]:
```suggestion
> configuration rather than to unexpectedly lose it.
```

File: content/docs/setup/kubernetes/minimal-install/index.md [hunk 11]
```diff
@@ -104,18 +103,20 @@ istio-pilot-58c65f74bc-2f5xn             1/1       Running   0          1m
...
 
-    If your Helm version is less than 2.10.0, then you need to manually cleanup extra job resource before redeploy new version of Istio chart:
-
-    {{< text bash >}}
-    $ kubectl -n istio-system delete job --all
-    {{< /text >}}
+* If desired, run the following command to delete all CRDs:
 
-* If desired, delete the CRDs:
+> {{< warning_icon >}} Deleting CRDs will delete any configuration changes that you have made to Istio.
```
Comment (rcaballeromx) [inline content/docs/setup/kubernetes/minimal-install/index.md]:
```suggestion
> {{< warning_icon >}} Deleting CRDs deletes any configuration changes that you have made to Istio.
```

File: content/docs/setup/kubernetes/minimal-install/index.md (hunk 2 above)
Comment (rcaballeromx) [reply content/docs/setup/kubernetes/minimal-install/index.md]: This recommendation is great because you then don't need to add them to the exceptions file.

File: .spelling [hunk 12]
```diff
@@ -530,6 +530,7 @@ subresources
 re-patch
 re-applied
 CustomResourceDefinitions
+CustomResources
```
[the remark this answers is not in the document]
Comment (morvencao) [reply .spelling]: Removed `CustomResources` from exception list.

File: content/docs/setup/kubernetes/minimal-install/index.md (hunk 7 above)
Comment (morvencao) [reply content/docs/setup/kubernetes/minimal-install/index.md]: done.

File: content/docs/setup/kubernetes/minimal-install/index.md (hunk 8 above)
Comment (morvencao) [reply content/docs/setup/kubernetes/minimal-install/index.md]: done.

File: content/docs/setup/kubernetes/minimal-install/index.md (hunk 9 above)
Comment (morvencao) [reply content/docs/setup/kubernetes/minimal-install/index.md]: done.

File: content/docs/setup/kubernetes/minimal-install/index.md (hunk 10 above)
Comment (morvencao) [reply content/docs/setup/kubernetes/minimal-install/index.md]: done

File: content/docs/setup/kubernetes/minimal-install/index.md (hunk 11 above)
Comment (morvencao) [reply content/docs/setup/kubernetes/minimal-install/index.md]: done

File: content/docs/setup/kubernetes/minimal-install/index.md [hunk 13]
```diff
@@ -15,21 +15,17 @@ Refer to the [prerequisites](/docs/setup/kubernetes/quick-start/#prerequisites)
...
 
 ### Option 1: Install with Helm via `helm template`
 
+Choose this option if your cluster doesn't have [Tiller](https://github.com/kubernetes/helm/blob/master/docs/architecture.md#components) deployed and you don't want to install it.
+
 1. Render Istio's core components to a Kubernetes manifest called `istio-minimal.yaml`:
 
     {{< text bash >}}
+    $ cat @install/kubernetes/namespace.yaml@ > $HOME/istio-minimal.yaml
+    $ cat install/kubernetes/helm/istio-init/files/crd-1* >> $HOME/istio-minimal.yaml
```
Comment (sdake) [inline content/docs/setup/kubernetes/minimal-install/index.md]: the naming convetion should be good with just `crd-*` - no need for the extra 1.

I'd also recommend against this installation method of creating the CRDs within the manifest - as if the manifest is deleted, all custom resources will be deleted.  Ideally CRD creation should be a separate step...  This is why we have an entirely seperate chart for the work.  That said, if an operator wishes to ensure the CRDs are applied without syncronizing on the completion of isito-init, that is straightforward

```
for $I in install/kubernetes/helm/istio-init/files/crd*; do kubectl apply -f $i; done
```

Note CRDs are cluster-scoped and do not require a namespace to be created.

File: content/docs/setup/kubernetes/minimal-install/index.md [hunk 14]
```diff
@@ -69,7 +64,13 @@ to manage the lifecycle of Istio.
     $ helm init --service-account tiller
     {{< /text >}}
 
-1. Install Istio:
+1. Install the `istio-init` chart to bootstrap all the Istio's [Custom Resource Definitions or CRDs for short](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/#customresourcedefinitions):
+
+    {{< text bash >}}
+    $ helm install install/kubernetes/helm/istio-init --name istio-init --namespace istio-system
+    {{< /text >}}
+
```
Comment (sdake) [inline content/docs/setup/kubernetes/minimal-install/index.md]: There should be a step about waiting for Istio's 58 crds to register.  On a new installation, I use this technique

```
kubectl get crds | wc -l
```
and if the value = 58, I know the CRD registration is done.  A better technique would be to check that all jobs created by istio-init are in completed 0/1 state.

File: content/docs/setup/kubernetes/minimal-install/index.md [hunk 15]
```diff
@@ -104,18 +105,20 @@ istio-pilot-58c65f74bc-2f5xn             1/1       Running   0          1m
...
-    $ kubectl -n istio-system delete job --all
-    {{< /text >}}
+* If desired, run the following command to delete all CRDs:
 
-* If desired, delete the CRDs:
+> {{< warning_icon >}} Deleting CRDs deletes any configuration changes that you have made to Istio.
 
     {{< text bash >}}
-    $ kubectl delete -f install/kubernetes/helm/istio/templates/crds.yaml
+    $ for i in install install/kubernetes/helm/istio-init/files/*crd-1*yaml; do kubectl delete -f $i; done
```
Comment (sdake) [inline content/docs/setup/kubernetes/minimal-install/index.md]: again `crd-*` should be sufficient.  We definately dont want a  wildcard on the beginning of the file.  the 1 is unnecessary.

File: content/docs/setup/kubernetes/minimal-install/index.md (hunk 6 above)
Comment (sdake) [reply content/docs/setup/kubernetes/minimal-install/index.md]: @rcaballeromx how about the following with some cleanup:
```
If you prefer to use Kubernetes manifests to deploy Istio, use option #1 below.  If you want Helm's TIller Pod(link to technical description of what TIller is) to manage your Istio release, please use Option 2 below.
```

File: content/docs/setup/kubernetes/minimal-install/index.md [hunk 16]
```diff
@@ -15,21 +15,17 @@ Refer to the [prerequisites](/docs/setup/kubernetes/quick-start/#prerequisites)
...
+Choose one of the following two **mutually exclusive** options described below.
 
 ### Option 1: Install with Helm via `helm template`
 
+Choose this option if your cluster doesn't have [Tiller](https://github.com/kubernetes/helm/blob/master/docs/architecture.md#components) deployed and you don't want to install it.
+
 1. Render Istio's core components to a Kubernetes manifest called `istio-minimal.yaml`:
 
     {{< text bash >}}
+    $ cat @install/kubernetes/namespace.yaml@ > $HOME/istio-minimal.yaml
```
Comment (sdake) [inline content/docs/setup/kubernetes/minimal-install/index.md]: this step is a pretty good improvement!

File: content/docs/setup/kubernetes/minimal-install/index.md [hunk 17]
```diff
@@ -15,21 +15,17 @@ Refer to the [prerequisites](/docs/setup/kubernetes/quick-start/#prerequisites)
...
+    $ cat install/kubernetes/helm/istio-init/files/crd-1* >> $HOME/istio-minimal.yaml
     $ helm template install/kubernetes/helm/istio --name istio --namespace istio-system \
       --set security.enabled=false \
       --set ingress.enabled=false \
```
Comment (ymesika) [inline content/docs/setup/kubernetes/minimal-install/index.md]: No need for this one because disabled by default

File: content/docs/setup/kubernetes/minimal-install/index.md (hunk 13 above)
Comment (morvencao) [reply content/docs/setup/kubernetes/minimal-install/index.md]: 1. My intention was that for minimal installation, the CRDs for `cert-manager` is not required. That's the reason why I added extra `1`.
2. Sounds good to create the CRDs in separate step, let me update that.

File: content/docs/setup/kubernetes/minimal-install/index.md (hunk 14 above)
Comment (morvencao) [reply content/docs/setup/kubernetes/minimal-install/index.md]: Done.

File: content/docs/setup/kubernetes/minimal-install/index.md (hunk 15 above)
Comment (morvencao) [reply content/docs/setup/kubernetes/minimal-install/index.md]: Done.

File: content/docs/setup/kubernetes/minimal-install/index.md (hunk 17 above)
Comment (morvencao) [reply content/docs/setup/kubernetes/minimal-install/index.md]: Nice catch!

File: content/docs/setup/kubernetes/minimal-install/index.md [hunk 18]
```diff
@@ -60,7 +50,7 @@ to manage the lifecycle of Istio.
 1. If a service account has not already been installed for Tiller, install one:
 
     {{< text bash >}}
-    $ kubectl apply -f install/kubernetes/helm/helm-service-account.yaml
+    $ kubectl apply -f @install/kubernetes/helm/helm-service-account.yaml@
```
Comment (linsun) [inline content/docs/setup/kubernetes/minimal-install/index.md]: what is the purpose of @ here?

File: content/docs/setup/kubernetes/minimal-install/index.md [hunk 19]
```diff
@@ -104,18 +96,20 @@ istio-pilot-58c65f74bc-2f5xn             1/1       Running   0          1m
...
-    $ kubectl -n istio-system delete job --all
-    {{< /text >}}
+* If desired, run the following command to delete all CRDs:
 
-* If desired, delete the CRDs:
+> {{< warning_icon >}} Deleting CRDs deletes any configuration changes that you have made to Istio.
 
     {{< text bash >}}
-    $ kubectl delete -f install/kubernetes/helm/istio/templates/crds.yaml
+    $ for i in install install/kubernetes/helm/istio-init/files/crd*yaml; do kubectl delete -f $i; done
```
Comment (linsun) [inline content/docs/setup/kubernetes/minimal-install/index.md]: extra install here?

File: content/docs/setup/kubernetes/minimal-install/index.md (hunk 18 above)
Comment (morvencao) [reply content/docs/setup/kubernetes/minimal-install/index.md]: Put @ on either end of the yaml file path, this will generate a link to the file on GitHub.

File: content/docs/setup/kubernetes/minimal-install/index.md [hunk 20]
```diff
@@ -15,39 +15,29 @@ Refer to the [prerequisites](/docs/setup/kubernetes/quick-start/#prerequisites)
...
 
-1. Choose one of the following two
-**mutually exclusive** options described below.
+Choose this option if your cluster doesn't have [Tiller](https://github.com/kubernetes/helm/blob/master/docs/architecture.md#components) deployed and you don't want to install it.
 
-### Option 1: Install with Helm via `helm template`
+1. Install all the Istio's [Custom Resource Definitions or CRDs for short](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/#customresourcedefinitions) via `kubectl apply`, and wait a few seconds for the CRDs to be committed in the Kube api-server:
+
+    {{< text bash >}}
+    $ for i in install/kubernetes/helm/istio-init/files/crd*yaml; do kubectl apply -f $i; done
```
Comment (frankbu) [inline content/docs/setup/kubernetes/minimal-install/index.md]: Is this approach really the right way to do it without helm? These CRDs have labels `heritage: Tiller` which makes no sense if Tiller is not used.

Shouldn't we have a build step that generates a non-helm yaml file (e.g., `istio-crds.yaml` that non-helm installs use. Don't we need that for users that install `istio-demo.yaml` anyway? https://preliminary.istio.io/docs/setup/kubernetes/quick-start/#option-1-install-istio-with-mutual-tls-enabled-and-set-to-use-permissive-mode-between-sidecars

File: content/docs/setup/kubernetes/minimal-install/index.md (hunk 19 above)
Comment (morvencao) [reply content/docs/setup/kubernetes/minimal-install/index.md]: Removed extra `install`.

File: content/docs/setup/kubernetes/minimal-install/index.md [hunk 21]
```diff
@@ -15,39 +15,29 @@ Refer to the [prerequisites](/docs/setup/kubernetes/quick-start/#prerequisites)
...
 
-1. Choose one of the following two
-**mutually exclusive** options described below.
+Choose this option if your cluster doesn't have [Tiller](https://github.com/kubernetes/helm/blob/master/docs/architecture.md#components) deployed and you don't want to install it.
 
-### Option 1: Install with Helm via `helm template`
+1. Install all the Istio's [Custom Resource Definitions or CRDs for short](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/#customresourcedefinitions) via `kubectl apply`, and wait a few seconds for the CRDs to be committed in the kube-apiserver:
+
+    {{< text bash >}}
+    $ for i in install install/kubernetes/helm/istio-init/files/crd*yaml; do kubectl apply -f $i; done
```
[the remark this answers is not in the document]
Comment (morvencao) [reply content/docs/setup/kubernetes/minimal-install/index.md]: Removed extra `install`.

File: content/docs/setup/kubernetes/minimal-install/index.md (hunk 20 above)
Comment (morvencao) [reply content/docs/setup/kubernetes/minimal-install/index.md]: @sdake Any thought for this?

Comment (morvencao) [reply content/docs/setup/kubernetes/minimal-install/index.md]: @frankbu Maybe we can open another PR for this, what do you think?

Comment (sdake) [reply content/docs/setup/kubernetes/minimal-install/index.md]: @frankbu currently istio-init.yaml is created by the build system.  In addition, when istio-demo.yaml is generated, it is a cat of namespace.yaml, the CRDs, followed by istio-demo.yaml.

We should do an audit of labels that were unnecessarily added to all of the resources in 1.1 and remove them.  @costinm had suggested this some time ago, but the work hasn't been done.  Once the labels are in, they cannot be removed as the label field in a spec is immutable and my understanding is labels are expensive in Kubernetes.

File: content/docs/setup/kubernetes/minimal-install/index.md (hunk 6 above)
Comment (sdake) [reply content/docs/setup/kubernetes/minimal-install/index.md]: @linsun ya that should probably be a `>` or an info emoji.

File: content/docs/setup/kubernetes/minimal-install/index.md (hunk 18 above)
Comment (sdake) [reply content/docs/setup/kubernetes/minimal-install/index.md]: yes I know, but why link it?

File: content/docs/setup/kubernetes/minimal-install/index.md [hunk 22]
```diff
@@ -69,21 +59,23 @@ to manage the lifecycle of Istio.
...
+1. Install the `istio-init` chart to bootstrap all the Istio's CRDs:
+
+    {{< text bash >}}
+    $ helm install install/kubernetes/helm/istio-init --name istio-init --namespace istio-system
+    {{< /text >}}
+
+1. Verify all the Istio's CRDs have been committed in the Kube api-server by checking all the CRD creation jobs complete with success:
+
+    {{< text bash >}}
+    $ kubectl get job --namespace istio-system | grep istio-crd
```
Comment (sdake) [inline content/docs/setup/kubernetes/minimal-install/index.md]: Are you sure this technique works?  I haven't tried it, however, the containers can exit and be in a 0/1 COMPLETED state, and still the CRDs lag in being commited in my experience.

File: content/docs/setup/kubernetes/minimal-install/index.md [hunk 23]
```diff
@@ -104,18 +96,20 @@ istio-pilot-58c65f74bc-2f5xn             1/1       Running   0          1m
 
 * For option 2, uninstall using Helm:
 
+> Uninstalling this chart does not delete Istio's registered CRDs. Istio, by design, expects
+> CRDs to leak into the Kubernetes environment. As CRDs contain all runtime configuration
+> data in custom resources the Istio designers consider better to explicitly delete this
```
Comment (sdake) [inline content/docs/setup/kubernetes/minimal-install/index.md]: `data in custom resource` -> `data for Istio configuraiton`.  This sentence is a little clunky.

File: content/docs/setup/kubernetes/minimal-install/index.md (hunk 20 above)
Comment (frankbu) [reply content/docs/setup/kubernetes/minimal-install/index.md]: @sdake, I guess you're saying that these are the correct instructions to install CRDs for the helm w/o tiller option. The istio-demo.yaml case is handled by istio-demo.yaml itself.

Is there an issue for removing `heritage: Tiller` (and any other unnecessary labels) from the templates? If I understand @sdake, it sounds like a must-have for 1.1.

File: content/docs/setup/kubernetes/minimal-install/index.md [hunk 24]
```diff
@@ -104,18 +96,20 @@ istio-pilot-58c65f74bc-2f5xn             1/1       Running   0          1m
...
-    $ kubectl -n istio-system delete job --all
-    {{< /text >}}
+* If desired, run the following command to delete all CRDs:
 
-* If desired, delete the CRDs:
+> {{< warning_icon >}} Deleting CRDs deletes any configuration changes that you have made to Istio.
 
     {{< text bash >}}
-    $ kubectl delete -f install/kubernetes/helm/istio/templates/crds.yaml
+    $ for i in install/kubernetes/helm/istio-init/files/crd*yaml; do kubectl delete -f $i; done
```
Comment (GregHanson) [inline content/docs/setup/kubernetes/minimal-install/index.md]: Is this the only cleanup step that needs to be done between istio installations?

File: content/docs/setup/kubernetes/minimal-install/index.md (hunk 22 above)
Comment (morvencao) [reply content/docs/setup/kubernetes/minimal-install/index.md]: How about we check the crds number from API-server?
```
kubectl get crds | grep 'istio.io\|certmanager.k8s.io' | wc -l 
```
The result is supposed to be `57`(`51` for `v1.0`, `3` for `v1.1`, `3` for `cert-manager`) for istio v1.1

File: content/docs/setup/kubernetes/minimal-install/index.md (hunk 24 above)
Comment (morvencao) [reply content/docs/setup/kubernetes/minimal-install/index.md]: Yes, this is the last step if users want to cleanup all istio `CRDs`. The namespace should be deleted if they choose `kubectl delete...` instead of `helm delete`. But `CRDs` are cluster-wide resources, so it doesn't matter if the namespace exists or not.

File: content/docs/setup/kubernetes/minimal-install/index.md (hunk 20 above)
Comment (sdake) [reply content/docs/setup/kubernetes/minimal-install/index.md]: @frankbu Right on.  As far as labels, we can't remove all of them - only the new ones that were introduced in release-1.1 branch.  Any present in the 1.0 branch need to stay, as labels are immutable and we wouldn't be able to upgrade.

I am unclear on the CRD labels - but I think the same applies.

Cheers
-steve

Comment (frankbu) [reply content/docs/setup/kubernetes/minimal-install/index.md]: Is somebody signed up to send a PR to remove these labels? @morvencao?

Comment (morvencao) [reply content/docs/setup/kubernetes/minimal-install/index.md]: @frankbu @sdake Since the labels can't be removed for hem /w tiller method, while they can for `kubectl apply ...` method. Can we add condition that only remove the labels for `kubectl apply...` and keep them for helm /w tiller?

Comment (morvencao) [reply content/docs/setup/kubernetes/minimal-install/index.md]: @sdake @frankbu Have raise a PR: https://github.com/istio/istio/pull/11301 for you review.

File: content/docs/setup/kubernetes/minimal-install/index.md (hunk 6 above)
Comment (rcaballeromx) [reply content/docs/setup/kubernetes/minimal-install/index.md]: I liked @sdake version because it clarifies under what circumstances one options should be chosen over the other without forcing the reader to scroll down to make the decision. Ideally, I would introduce the options in a list and link to the appropriate heading in the text. One possibility would be:
```suggestion
You have two mutually exclusive options to install Istio:
- To use Kubernetes manifests to deploy Istio,  follow the instructions for [option #1](#option1).
- To use [Helm's Tiller pod](link to technical description of what TIller is) to manage your Istio release, follow the instructions for [option #2](#option2).
```

File: content/docs/setup/kubernetes/minimal-install/index.md [hunk 25]
```diff
@@ -15,39 +15,29 @@ Refer to the [prerequisites](/docs/setup/kubernetes/quick-start/#prerequisites)
...
 ## Installation steps
 
-1. If using a Helm version prior to 2.10.0, install Istio's [Custom Resource Definitions](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/#customresourcedefinitions)
-via `kubectl apply`, and wait a few seconds for the CRDs to be committed to the Kubernetes API server:
+Choose one of the following two **mutually exclusive** options described below.
 
-    {{< text bash >}}
-    $ kubectl apply -f install/kubernetes/helm/istio/templates/crds.yaml
-    {{< /text >}}
+### Option 1: Install with Helm via `helm template`
```
Comment (rcaballeromx) [inline content/docs/setup/kubernetes/minimal-install/index.md]: For the suggested links to work, I recommend adding a label to this heading:
```suggestion
### Option 1: Install with Helm via `helm template` {#option1}
```

File: content/docs/setup/kubernetes/minimal-install/index.md [hunk 26]
```diff
@@ -15,39 +15,29 @@ Refer to the [prerequisites](/docs/setup/kubernetes/quick-start/#prerequisites)
...
-      --set pilot.sidecar=false > $HOME/istio-minimal.yaml
+      --values install/kubernetes/helm/istio/values-istio-minimal.yaml >> $HOME/istio-minimal.yaml
     {{< /text >}}
 
 1. Install the Pilot component via the manifest:
 
     {{< text bash >}}
-    $ kubectl create namespace istio-system
     $ kubectl apply -f $HOME/istio-minimal.yaml
     {{< /text >}}
```
Comment (rcaballeromx) [inline content/docs/setup/kubernetes/minimal-install/index.md]: You need to change the heading for the link at the top of the file to work.

### Option 2: Install with Helm and Tiller via `helm install` {#option2}

**Comment by morvencao** [comment]
@rcaballeromx Applied your comments, go ahead for next pass.

**Comment by rcaballeromx** [comment]
@morvencao I won't manage to do it today but you are first in my queue Monday.

File: content/docs/setup/kubernetes/minimal-install/index.md [hunk 27]
```diff
@@ -69,21 +61,23 @@ to manage the lifecycle of Istio.
...
 
-1. Install Istio:
+1. Install the `istio-init` chart to bootstrap all the Istio's CRDs:
+
+    {{< text bash >}}
+    $ helm install install/kubernetes/helm/istio-init --name istio-init --namespace istio-system
+    {{< /text >}}
+
+1. Verify all the Istio's CRDs have been committed in the Kube api-server by checking the number of CRD with specified suffix. Run the following command and check the result, totally `57` CRDs should be created for Istio:
```
Comment (rcaballeromx) [inline content/docs/setup/kubernetes/minimal-install/index.md]: Mind the line length. The text is a bit hard to follow due to all the different tenses present. Try simplifying the text to make it easier to understand:
```suggestion
1. To verify all Istio's CRDs were committed in the Kube api-server, check the number of CRDs with the specified suffix with the following command. Verify that the number of total CRDs created was `57` for Istio:
```

File: content/docs/setup/kubernetes/minimal-install/index.md [hunk 28]
```diff
@@ -104,18 +98,20 @@ istio-pilot-58c65f74bc-2f5xn             1/1       Running   0          1m
 
 * For option 2, uninstall using Helm:
 
+> Uninstalling this chart does not delete Istio's registered CRDs. Istio, by design, expects
+> CRDs to leak into the Kubernetes environment. As CRDs contain all runtime configuration
+> data for Istio configuration the Istio designers consider better to explicitly delete this
+> configuration rather then unexpectedly to lose it.
```
Comment (rcaballeromx) [inline content/docs/setup/kubernetes/minimal-install/index.md]: Typo:
```suggestion
> configuration rather than unexpectedly to lose it.
```

File: content/docs/setup/kubernetes/minimal-install/index.md [hunk 29]
```diff
@@ -15,43 +15,35 @@ Refer to the [prerequisites](/docs/setup/kubernetes/quick-start/#prerequisites)
...
 
 1. Install the Pilot component via the manifest:
 
     {{< text bash >}}
-    $ kubectl create namespace istio-system
     $ kubectl apply -f $HOME/istio-minimal.yaml
     {{< /text >}}
 
-### Option 2: Install with Helm and Tiller via `helm install`
+### Option 2: Install with Helm and Tiller via `helm install` {#option2}
```
Comment (sdake) [inline content/docs/setup/kubernetes/minimal-install/index.md]: not sure what #option1 and #option2 do.  The links work without it, do they not?

File: .spelling [hunk 30]
```diff
@@ -318,6 +318,8 @@ OpenID_Connect
 OpenShift
 OpenSSL
 openssl
+option1
```
Comment (sdake) [inline .spelling]: this is troubling.  Each time we add a link, we don't want to add that to  the .spelling file.

@geeknoid any offer for a solution to this problem?

File: content/docs/setup/kubernetes/minimal-install/index.md (hunk 27 above)
Comment (sdake) [reply content/docs/setup/kubernetes/minimal-install/index.md]: I would change `Kube api-server to "Kubernetes api-server`.  I am also not clear that 57 is accurate.  On a fresh cluster, I get 59:

kubectl get crds | grep 'istio.io\|certmanager.k8s.io' | wc -l

File: content/docs/setup/kubernetes/minimal-install/index.md [hunk 31]
```diff
@@ -104,18 +98,20 @@ istio-pilot-58c65f74bc-2f5xn             1/1       Running   0          1m
 
 * For option 2, uninstall using Helm:
 
+> Uninstalling this chart does not delete Istio's registered CRDs. Istio, by design, expects
+> CRDs to leak into the Kubernetes environment. As CRDs contain all runtime configuration
+> data for Istio configuration the Istio designers consider better to explicitly delete this
```
Comment (sdake) [inline content/docs/setup/kubernetes/minimal-install/index.md]: This sentence is klunky.  Recommend breaking it up with periods.  `CRDs contain all runtime configuration for Istio configuration.  The Istio designers consider it better for operators to explicitly delete the runtime configuration rather than unexpectedly lose it.

File: content/docs/setup/kubernetes/minimal-install/index.md (hunk 24 above)
Comment (sdake) [reply content/docs/setup/kubernetes/minimal-install/index.md]: @GregHanson - depends on the workflow.  For option 1, the namespace needs to be deleted.  For option 2, the namespace is automatically deleted.

File: content/docs/setup/kubernetes/minimal-install/index.md (hunk 29 above)
Comment (rcaballeromx) [reply content/docs/setup/kubernetes/minimal-install/index.md]: They do if you include the entire heading on the link. Adding the label `{#option2}` makes the link easier to maintain because the heading can change without breaking the link.

File: .spelling (hunk 30 above)
Comment (rcaballeromx) [reply .spelling]: I would suggest changing the label to option-1 to avoid problems with the spellchecker.
```suggestion
```

File: content/docs/setup/kubernetes/minimal-install/index.md (hunk 31 above)
Comment (rcaballeromx) [reply content/docs/setup/kubernetes/minimal-install/index.md]: I agree, what about:
```suggestion
CRDs contain all the runtime configuration data needed to configure Istio. Because of this, we consider it better for operators to explicitly delete the runtime configuration data rather than unexpectedly lose it.
```

Comment (morvencao) [reply content/docs/setup/kubernetes/minimal-install/index.md]: Done

File: .spelling (hunk 30 above)
Comment (morvencao) [reply .spelling]: Change to `option-1` and `option-2`

File: content/docs/setup/kubernetes/minimal-install/index.md (hunk 28 above)
Comment (morvencao) [reply content/docs/setup/kubernetes/minimal-install/index.md]: Done

File: content/docs/setup/kubernetes/minimal-install/index.md (hunk 27 above)
Comment (morvencao) [reply content/docs/setup/kubernetes/minimal-install/index.md]: @sdake Check CRD number again with latest `release-1.1`:
49 CRDs in https://github.com/istio/istio/blob/release-1.1/install/kubernetes/helm/istio-init/files/crd-10.yaml
4 CRDs in https://github.com/istio/istio/blob/release-1.1/install/kubernetes/helm/istio-init/files/crd-11.yaml
3 CRDs in https://github.com/istio/istio/blob/release-1.1/install/kubernetes/helm/istio-init/files/crd-certmanager-10.yaml
